accounting/accounting_cli_functions.py

```diff
@@ -168,76 +168,52 @@ def add_job_records(dataframe):
         else:
-            write_records_to_file(job_records, output_file)
-
-    return job_records
-
-
-def view_jobs_with_jobid(conn, jobid, output_file):
-    # get the information pertaining to a job in the jobs DB
-    select_stmt = (
-        "SELECT userid,id,t_submit,t_run,t_inactive,ranks,R FROM jobs where id=?"
-    )
-    dataframe = pd.read_sql_query(select_stmt, conn, params=(jobid,))
-    # if the length of dataframe is 0, that means
-    # the jobid specified was not found in the table
-    if len(dataframe.index) == 0:
-        return "Job not found in jobs table"
-    else:
-        job_records = add_job_records(dataframe)
-        if output_file is None:
-            print_job_records(job_records)
+            where_stmt += "WHERE userid=? "
+    if "after_start_time" in params:
+        params_list.append(params["after_start_time"])
+        if where_stmt != "":
+            where_stmt += "AND t_run > ? "
         else:
-            write_records_to_file(job_records, output_file)
-
-    return job_records
-
-
-def view_jobs_after_start_time(conn, time_after, output_file):
-    # get jobs that have completed after a certain time
-    select_stmt = (
-        "SELECT userid,id,t_submit,t_run,t_inactive,ranks,R FROM jobs WHERE t_run > ?"
-    )
-    dataframe = pd.read_sql_query(select_stmt, conn, params=(time_after,))
-    # if the length of dataframe is 0, that means
-    # the time specified resulted in no jobs found
-    if len(dataframe.index) == 0:
-        return "No jobs found after time specified"
-    else:
-        job_records = add_job_records(dataframe)
-        if output_file is None:
-            print_job_records(job_records)
+            where_stmt += "WHERE t_run > ? "
+    if "before_end_time" in params:
+        params_list.append(params["before_end_time"])
+        if where_stmt != "":
+            where_stmt += "AND t_inactive < ? "
         else:
-            write_records_to_file(job_records, output_file)
-
-    return job_records
+            where_stmt += "WHERE t_inactive < ? "
+    if "jobid" in params:
+        params_list.append(params["jobid"])
+        if where_stmt != "":
+            where_stmt += "AND id=? "
+        else:
+            where_stmt += "WHERE id=? "
 
+    select_stmt += where_stmt
 
-def view_jobs_before_end_time(conn, time_before, output_file):
-    # get jobs that have completed before a certain time
-    select_stmt = "SELECT userid,id,t_submit,t_run,t_inactive,ranks,R FROM jobs WHERE t_inactive < ?"
-    dataframe = pd.read_sql_query(select_stmt, conn, params=(time_before,))
+    dataframe = pd.read_sql_query(select_stmt, conn, params=((*tuple(params_list),)))
     # if the length of dataframe is 0, that means
-    # the time specified resulted in no jobs found
+    # no job records were found in the jobs table
     if len(dataframe.index) == 0:
-        return "No jobs found before time specified"
+        return "Record(s) not found in jobs table"
```

Comment:
although you should defer to any users you are talking to, my preference would be to have no output if no records match.  But that's me.

Reply:
I like this suggestion - thanks. No output would actually be more in line with SQLite's behavior, which I believe just returns an empty `Cursor` object:

```
sqlite> select * from jobs where user='bad_username'
sqlite> 
``` 

In my specific case, I could probably just return an empty `job_records` list.